pulser-core/pulser/sequence/sequence.py

```diff
@@ -231,44 +236,39 @@ def declared_variables(self) -> dict[str, Variable]:
     def available_channels(self) -> dict[str, Channel]:
         """Channels still available for declaration."""
         all_channels = {**self._device.channels, **self._device.dmm_channels}
-        xy_channels = {
-            id: ch for id, ch in all_channels.items() if ch.basis == "XY"
-        }
-        if not self._schedule:
-            # Show all channels if none are declared
-            if not self._in_xy:
-                return all_channels
-            # If in xy_mode, the DMMs are available if no SLM has been defined
-            elif not self._slm_mask_dmm:
-                return {**xy_channels, **self._device.dmm_channels}
-            return xy_channels
+        if not self._schedule and not self._in_xy and not self._in_ising:
+            # If no channel has been declared nor any DMM configured
+            # Take out the DMM for the SLM Mask if channels are not reusable
```

Comment:
Perhaps `Don't show` would be better than `Take out` ?
```suggestion
            # If no channel has been declared nor any DMM configured, and if channels 
            # are not reusable, don't show the DMM used for the SLM Mask 
```

Reply:
Yes, I agree